Libraries/LibJS/Runtime/ObjectConstructor.cpp

```diff
@@ -104,6 +105,19 @@ Value ObjectConstructor::set_prototype_of(Interpreter& interpreter)
     return {};
 }
 
+
+Value ObjectConstructor::prevent_extensions(Interpreter& interpreter)
+{
+    auto argument = interpreter.argument(0);
+    if (!argument.is_object())
+        return argument;
+    if (!argument.as_object().prevent_extensions()) {
+        interpreter.throw_exception<TypeError>("Proxy preventExtensions handler returned false");
```

Comment:
Just a note here:

This might look like a really strange error message, but from what I can tell by looking through the spec, the _only_ way a `prevent_extensions()` call can return false is if a user defines a Proxy trap that returns false. Plus I'm working on Proxies right now :) so it'll be less weird in a couple days.